src/TempleLineOfCredit.sol

```diff
@@ -0,0 +1,318 @@
+        interestRatePeriod = _interestRatePeriod;
+        
+        collateralToken = IERC20(_collateralToken);
+        collateralPrice = _collateralPrice;
+
+        debtToken = IERC20(_debtToken);
+        debtPrice = _debtPrice;
+        
+        debtCollector = _debtCollector;
+    }
+
+
+    function addOperator(address _address) external override onlyOwner {
+        _addOperator(_address);
+    }
+
+    function removeOperator(address _address) external override onlyOwner {
+        _removeOperator(_address);
+    }
+
+    function setDebtPrice(uint256 _debtPrice) external onlyOperators {
+        debtPrice = _debtPrice;
+    }
+
+    function setCollateralPrice(uint256 _collateralPrice) external onlyOperators {
+        collateralPrice = _collateralPrice;
+    }
+
+    function setDebtCollector(address _debtCollector) external onlyOperators {
+        debtCollector = _debtCollector;
+    }
+
+    function setCollateralizationRatio(uint256 _minCollateralizationRatio) external onlyOperators {
+        minCollateralizationRatio = _minCollateralizationRatio;
+    }
+
+    /**
+     * @dev Get user principal amount
+     * @return principal amount
+     */
+    function getDebtAmount(address account) public view returns (uint256) {
+        return positions[account].debtAmount;
+    }
+
+    /**
+     * @dev Get user total debt incurred (principal + interest)
+     * @return total Debt
+     */
+    function getTotalDebtAmount(address account) public view returns (uint256) {
+        uint256 totalDebt = positions[account].debtAmount;
+        uint256 periodsPerYear = 365 days / interestRatePeriod;
+        uint256 periodsElapsed = block.timestamp - positions[account].createdAt; // divided by interestRatePeriod
+        totalDebt += (((totalDebt * interestRateBps) / 10000 / periodsPerYear) * periodsElapsed) / interestRatePeriod;
+        return totalDebt;
+    }
+
+    /**
+     * @dev Allows operator to depoist debt tokens
+     * @param amount is the amount to deposit
+     */
+    function depositDebt(address account, uint256 amount) external onlyOperators{
+        if (amount == 0) {
+            revert InvalidAmount(amount);
+        }
+        debtBalance += amount;
+        debtToken.safeTransferFrom(
+            account,
+            address(this),
+            amount
+        );
+        emit DepositDebt(amount);
+    }
+
+        /**
+     * @dev Allows operator to remove debt token
+     * @param amount is the amount to remove
+     */
+    function removeDebt(uint256 amount) external onlyOperators{
+        if (amount == 0) {
+            revert InvalidAmount(amount);
+        }
+        debtBalance -= amount;
+        debtToken.safeTransfer(
+            msg.sender,
+            amount
+        );
+        emit RemoveDebt(amount);
+    }
+
+    /**
+     * @dev Allows borrower to deposit collateral
+     * @param amount is the amount to deposit
+     */
+    function postCollateral(uint256 amount) external {
```

Comment:
Wonder if we should be cautious and mark nonReentrant just to be sure. Eg any flash loan scenarios which could post `collateral->borrow->repay`?